core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala

```diff
@@ -158,6 +159,20 @@ private[backends] trait SharedBackendUtils extends Logging with Serializable {
       .buildArgs()
   }
 
+  def parseStringToHttpHeaderArgs(headers: String): Seq[String] = {
+    val headerPattern = """^\s*([^:]+)\:\s*(.+)$""".r
+    headers.split('\n').flatMap{ header =>
```

Comment:
Minor: I'd suggest pre-compiling the resulting finite automaton for performance reasons, but it's not really crucial in this pase. Would personally use pre-compiled `scala.util.matching.Regex`.

Reply:
As per personal discussion, I will leave it as it's since regexp is used only once during the application run.